op-batcher/batcher/driver_test.go

```diff
@@ -117,3 +142,92 @@ func TestBatchSubmitter_SafeL1Origin_FailsToResolveRollupClient(t *testing.T) {
 	_, err := bs.safeL1Origin(context.Background())
 	require.Error(t, err)
 }
+
+func TestBatchSubmitter_CheckRecentTxsOnStart(t *testing.T) {
+	bs, ep := setup(t)
+	txMgr := bs.Txmgr.(*mocks.TxManager)
+	l1Client := bs.L1Client.(*MockL1Client)
```

Comment:
Maybe we move these into `t.Run` to set up the batch submitter and mocks for individual tests? Then we don't need to reset mock fields and are guaranteed to have fresh mocks and a fresh BS. Note that if we move the recent trx logic into its own function `CheckRecentTxs`, the setup becomes a bit lighter.